### PR TITLE
Make the VortexReadAt::size method return an io::Result

### PR DESCRIPTION
**Pull request by joseph-isaacs**

### CODE REVIEW

File: vortex-io/src/tokio.rs [hunk 1]
```diff
@@ -81,18 +81,17 @@ impl VortexReadAt for TokioFile {
...
-    fn size(&self) -> impl Future<Output = u64> + 'static {
+    fn size(&self) -> impl Future<Output = io::Result<u64>> + 'static {
         let this = self.clone();
 
         async move {
-            let res = tokio::task::spawn_blocking(move || {
+            tokio::task::spawn_blocking(move || {
                 this.metadata()
                     .unwrap_or_else(|e| vortex_panic!("access TokioFile metadata: {e}"))
                     .len()
```
Comment (a10y) [inline vortex-io/src/tokio.rs]: we can just forward the io::result from this.metadata()

Comment (joseph-isaacs) [reply vortex-io/src/tokio.rs]: Good spot

**Review by a10y — APPROVED** [review]
feel free to merge once tests pass

File: vortex-io/src/tokio.rs [hunk 2]
```diff
@@ -81,17 +81,17 @@ impl VortexReadAt for TokioFile {
...
                 this.metadata()
-                    .unwrap_or_else(|e| vortex_panic!("access TokioFile metadata: {e}"))
-                    .len()
+                    .map(|metadata| metadata.len())
             })
-            .await
-            .map_err(io::Error::other)
+                .await
+                .map_err(io::Error::other)
+                .and_then(identity)
```
Comment (a10y) [inline vortex-io/src/tokio.rs]: honestly we can probably remove the spawn_blocking, doing an fstat should be pretty much instantaneous

**Pull request merged**